examples/MNIST_classification.ipynb

```diff
@@ -0,0 +1,598 @@
+    "plot_heatmap(X_train[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Binarization of the images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "binarizer = Binarizer(threshold=0.4)\n",
+    "X_train_binarized = binarizer.fit_transform(X_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# NOTE TO UMBE: plot_heatmap does not support binary arrays\n",
+    "plot_heatmap(X_train_binarized[1]*1.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Inverting the boolean images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inverter = Inverter(n_jobs=4)\n",
+    "X_train_inverted = inverter.fit_transform(X_train_binarized)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_heatmap(X_train_inverted[1]*1.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Applying a boolean image filtration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_iterations = 28\n",
+    "\n",
+    "erosion_filtration = ErosionFiltration(n_iterations=n_iterations, n_jobs=4)\n",
+    "X_train_filtered = erosion_filtration.fit_transform(X_train_inverted)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_heatmap(X_train_filtered[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Getting persistence diagrams out of images"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cubical_complex = CubicalPersistence(n_jobs=1)\n",
+    "X_train_cubical = cubical_complex.fit_transform(X_train_filtered)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plot_diagram(X_train_cubical[1])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Computing the betti curves"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "betti = BettiCurve(n_bins=36, n_jobs=1)\n",
+    "X_train_betti = betti.fit_transform(X_train_cubical)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "betti.plot(X_train_betti, sample=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Computing the heat kernel of stacked diagrams"
```

Comment:
Maybe let's call it "the heat kernel vector representation". We have a longstanding problem with the naming of this transformer, which doesn't actually compute a kernel...

Reply:
What I meant here is to do the heat kernel of the collection of diagrams for which dimensions have been "forgotten".

But I agree with the naming, It is not much more "heat" than the persistence image either. But that's the name it is given in the literature...

Reply:
My issue is with the word "kernel". What we call "heat kernel" is not in fact a kernel. A kernel is a function of two arguments so would better correspond to our distance matrix based on this heat representation. There is a long thread somewhere with @ammedmar and @wreise , let me find this.

Reply:
https://github.com/giotto-ai/giotto-tda/pull/348#discussion_r390156070

Unfortunately, no action was ever taken, but it should have been!

Reply:
Oops I completely misunderstood your comment, I updated the naming as you recommended.